Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,163 @@
+    LogMsg "Setup network trace"
+	watch -n 5 curl https://microsoft.com &>/dev/null &
+	watch_pid=$!
+}
+
+function enable_disable_da(){
+	verify_install_da
+	
+	sleep 30
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/service.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log does not exist." && exit 1
+
+	service_log_time=$(date -r /var/opt/microsoft/dependency-agent/log/service.log +%s)
+	current_time=$(date +%s)
+	time_elapsed=$(($current_time - $service_log_time))
+	sleep $((120-$time_elapsed))
+
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.1 does not exist." && exit 1
+	[ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && LogErr "/var/opt/microsoft/dependency-agent/log/service.log.2 exist." && exit 1
+	[ ! -c "/dev/msda" ] && LogErr "/dev/msda does not exist." && exit 1
+
+	! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && LogErr "driver setup status=0 not found" && exit 1
+	! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && LogErr "starting the dependency agent not found" && exit 1
+	
+	sleep 60
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log does not exist." && exit 1
+	[ ! -f "/etc/opt/microsoft/dependency-agent/config/DA_PID" ] && LogErr "/etc/opt/microsoft/dependency-agent/config/DA_PID does not exist." && exit 1
+	
+	if grep -iq "^[1-9][0-9]*$" /etc/opt/microsoft/dependency-agent/config/DA_PID; then
+        x=$(grep -i "^[1-9][0-9]*$" /etc/opt/microsoft/dependency-agent/config/DA_PID)
+        strings /proc/$x/cmdline | grep "/opt/microsoft/dependency-agent/bin/microsoft-dependency-agent-manager" && LogMsg "PID matched" || (LogErr "PID not matched" && exit 1)
+	else
+		LogErr "PID not found in DA_PID"
+		exit 1 
+	fi
+	
+	sleep 120
+	! ls /var/opt/microsoft/dependency-agent/storage/*.bb && LogErr "/var/opt/microsoft/dependency-agent/storage/*.bb does not exist." && exit 1
+
+	sleep 90
+	! ls /var/opt/microsoft/dependency-agent/storage/*.hb && LogErr "/var/opt/microsoft/dependency-agent/storage/*.hb does not exist." && exit 1
+	[ ! -f "/etc/opt/microsoft/dependency-agent/config/DA_PID" ] && LogErr "/etc/opt/microsoft/dependency-agent/config/DA_PID does not exist." && exit 1
+
+	if grep -iq "^[1-9][0-9]*$" /etc/opt/microsoft/dependency-agent/config/DA_PID; then
+        x=$(grep -i "^[1-9][0-9]*$" /etc/opt/microsoft/dependency-agent/config/DA_PID)
+        strings /proc/$x/cmdline | grep "/opt/microsoft/dependency-agent/bin/microsoft-dependency-agent-manager" && LogMsg "PID matched" || (LogErr "PID not matched" && exit 1)
+	else
+		LogErr "PID not found in DA_PID"
+		exit 1 
+	fi
+
+	dmesg | egrep -w "BUG:|Modules linked in:|Call Trace:​" && LogErr "Found BUG/Modules linked in/Call Trace in dmesg" && exit 1
```

Comment:
Will the output of the `egrep` be preserved where it will be accessible to the person troubleshooting a test failure?
Since these are fixed strings, consider using `fgrep -e BUG: -e 'Modules linked in:' ...`.
Consider adding some additional lines of context around the match. See "Context Line Control" in the grep man page.

Reply:
Changed to fgrep